cloudinit/sources/helpers/aeza.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (C) 2024 Aeza.net.
+#
+# Author: Egor Ternovoy <cofob@riseup.net>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import requests
+import time
+from requests.exceptions import HTTPError, RequestException
+
+from cloudinit import util, dmi
+
+
+def read_url(url, timeout=None, sec_between=1, retries=0):
```

Comment:
Let's not reinvent read_url helpers if we can avoid it as it makes it harder to maintain utility code that evolves over time. Is there something that [cloudinit.url_helper.readurl](https://github.com/canonical/cloud-init/blob/main/cloudinit/url_helper.py#L353) doesn't do for you?

Reply:
Also it doesn't support `max_wait` parameter.